frontends/web/src/components/layout/version.tsx

```diff
@@ -14,19 +15,32 @@
  * limitations under the License.
  */
 
-import { PropsWithChildren } from 'react';
-import { load } from '../../decorators/load';
-import { translate, TranslateProps } from '../../decorators/translate';
+import { FunctionComponent, useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { getVersion } from '../../api/version';
 
-interface VersionProps {
-    version: string;
-}
+const Version: FunctionComponent = () => {
+    const { t } = useTranslation();
+    const [version, setVersion] = useState<string>('');
 
-type Props = VersionProps & TranslateProps;
+    useEffect(() => {
+        let ignore = false;
+        async function updateVersion() {
+            const result = await getVersion();
+            if (!ignore) {
+                setVersion(result);
+            }
+        }
+        updateVersion();
+        return () => {
+            ignore = true
+        };
+    });
 
-function Version({ t, version }: PropsWithChildren<Props>) {
+    if (!version) {
```

Comment:
If we define a default loading string like `...` for version this check would not be needed, do you think it's better?

Reply:
In Fact, the API responses are so fast that even with an empty string would be unnoticeable

Reply:
yeah true or just a "loading" message 🤷‍♂️  , in the future we could have the gray loading indicator, but this is a bit out of scope for the react-migration.

I believe the version would load faster, but if I remember correctly it is still inside 2 components that have the load decorator I believe that is why it is taking so long to see the version (about .5s)

The missing version text doesn't make the page layout shift, it just appears once the version is loaded, so I would keep it as is for now.

Reply:
As far as rendering a null doesn't break the Layout (not this case), im ok, this is just a suggestion :+1: